#########################################################################
# joint control PD
#########################################################################
PD:
  joint_kp: [800, 800, 800, 800, 300, 50, 10]
  joint_kd: [80, 100, 80, 80, 10, 1, 1]

  joint_kp_joint_impedance: [600, 600, 500, 500, 150, 12, 12.5]
  joint_kd_joint_impedance: [120, 120, 80,  80,  30,  12.5, 12]

freq: 200

max_torque: [175, 110, 110, 80, 110, 40, 40]

gravity: 19.62
#gravity: 0.0
center_ee: [0.01018, 0.01031, 0.05839]
#########################################################################
# Pre-defined pose
#########################################################################

_iiwa_home: [-0.32032434, 0.02706913, -0.22881953, -1.42621454, 1.3862661, 0.55966738, 1.79477984 - 3.14159265]
_iiwa_home_pose: [0.47769025, -0.21113556, 0.78239543, 0.70487697, 0.00342266, 0.70931291, 0.0034541]

hand_home_pose: [ 0.03063398, -0.15441399,  0.82104033,  0.82512679,  0.00664029,
       -0.2039733 ,  0.80409743,  0.        , -0.10271061, -0.06070025,
        0.90650362,  0.81377007,  0.86300686,  0.44554398,  0.08874171,
        0.77574977]
hand_catch_pose: [-0.50889807,  1.74874918,  0.12737114,  0.39226041,  0.07008444,
        1.64808905,  1.36754356,  0.21940653,  0.0937576 ,  1.77053021,
        1.36580765,  0.55991777,  0.94150295,  0.14131779, -0.24060529,
        0.96755097]

# training
envelop_pose: [ 0.3794, 0.1122, 0.8395, 0.2504, -0.5161, 0.7940, 0.9335,
                0.5431, -0.4874, 0.3151, 0.5176, 0.4811, 1.0792, 0.2427,
                0.2730, 1.1701]

exp_envelop_pose: [ 0.36612518,  0.45032315,  0.71964249,  0.11512009, -0.14777576,
        1.23505241,  0.78363779,  0.44444217, -0.57235144,  0.48888354,
        0.63816652,  0.34616258,  1.36452549,  0.52015403,  0.68981928,
        1.27919653]

release_pose_A: [ 0.49417469,  0.4324724 ,  0.58722069,  0.08109998, -0.31399671,
        0.9837246 ,  0.84353987,  0.16538824, -0.56719455,  0.40209367,
        0.56480713,  0.3540265 ,  1.17035376,  0.61247766, -0.19364582,
        0.90379205]

release_pose_B: [ 0.06062334, -0.28774685,  0.58036132, -0.27871771, -0.12211313,
        1.24530951,  0.79705047,  0.4162993 , -0.3210622 , -0.0840534 ,
        0.24383255,  0.22312064,  1.39094669,  0.48692642,  0.68869331,
        1.29376476]

#########################################################################
# Trajectory generation
#########################################################################

ERROR_THRESHOLD: 0.2  # Threshold to switch from homing to throwing state
GRIPPER_DELAY: 0.04

# Ruckig margins for throwing
MARGIN_VELOCITY: 1

#MARGIN_ACCELERATION: 0.5
#MARGIN_JERK: 0.008

# Multi-target
MARGIN_ACCELERATION: 0.25
MARGIN_JERK: 0.005


# constraints of iiwa 7
# joint 1 and joint 4 has lower max_velocity because of compensation of gravity
max_velocity: [1.71, 1.74, 1.745, 2.269, 2.443, 3.142, 3.142]
max_acceleration: [3, 3, 7.5, 12.5, 15, 25, 20]
max_jerk: [1500, 300, 4000, 6000, 7500, 10000, 10000]

# for simulation
#max_acceleration: [15, 15, 7.5, 12.5, 15, 25, 20]
#max_jerk: [6000, 6000, 4000, 6000, 7500, 10000, 10000]


#########################################################################
# Matching parameters
#########################################################################
# real robot
thres_dis: 0.01
thres_v: 0.1
thres_r_ratio: 0.8
thres_height: 0.1
thres_phi_ratio: 0.8

# for training
#thres_dis: 0.01
#thres_v: 0.1
#thres_r_ratio: 0.7
#thres_height: 0.12
#thres_phi_ratio: 0.7


